pkg/search: TestQuery test flake logging in goroutine

Looks like a flaky test that's logging to `t.Logf` from a goroutine.

Should add correct synchronization waiting for things to shut down in the test, or maybe something like https://pkg.go.dev/tailscale.com/tstest#WhileTestRunningLogger as a crutch

```
% go test -count=5 ./pkg/search
...
2025/05/27 14:20:39 index/corpus: scanning CPU usage: 0s
--- FAIL: TestQuery (0.01s)
    query_test.go:109: Testing: --querytype=classic ...
panic: Log in goroutine after TestIsCheckinQuerySource has completed: unexpected candidateSource: got index_blob_meta, want corpus_permanode_types [recovered]
        panic: Log in goroutine after TestIsCheckinQuerySource has completed: unexpected candidateSource: got index_blob_meta, want corpus_permanode_types

goroutine 3593 [running]:
testing.tRunner.func1.2({0x100831240, 0x14000840590})
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/testing/testing.go:1734 +0x1ac
testing.tRunner.func1()
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/testing/testing.go:1737 +0x334
panic({0x100831240?, 0x14000840590?})
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/runtime/panic.go:792 +0x124
testing.(*common).logDepth(0x140008e2000, {0x14000dd2d20, 0x4c}, 0x3)
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/testing/testing.go:1064 +0x3bc
testing.(*common).log(...)
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/testing/testing.go:1046
testing.(*common).Fatalf(0x140008e2000, {0x10070e11c?, 0x14000f9c550?}, {0x14001884480?, 0x140000a7848?, 0x100670b6c?})
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/testing/testing.go:1124 +0x5c
perkeep.org/pkg/search_test.(*queryTest).wantRes.func1({0x1006f980e, 0xf})
        /Users/bradfitz/src/perkeep.org/pkg/search/query_test.go:156 +0xfc
perkeep.org/pkg/search.(*Handler).Query(0x140012f2200, {0x1008e6698?, 0x100c124a0?}, 0x14000f9c410)
        /Users/bradfitz/src/perkeep.org/pkg/search/query.go:1014 +0x350
perkeep.org/pkg/search_test.(*queryTest).wantRes(0x1400196e360, 0x14000f9c410, {0x140000a7e08, 0x2, 0xb?})
        /Users/bradfitz/src/perkeep.org/pkg/search/query_test.go:160 +0xd0
perkeep.org/pkg/search_test.TestQuery.func1(0x1400196e360)
        /Users/bradfitz/src/perkeep.org/pkg/search/query_test.go:193 +0xf4
perkeep.org/pkg/search_test.testQueryType({0x1008ea280, 0x14000bee000}, 0x1008dcff0, 0x0)
        /Users/bradfitz/src/perkeep.org/pkg/search/query_test.go:146 +0x2b0
perkeep.org/pkg/search_test.testQueryTypes({0x1008ea280, 0x14000bee000}, {0x100b75b50, 0x3, 0x14001c48f58?}, 0x1008dcff0)
        /Users/bradfitz/src/perkeep.org/pkg/search/query_test.go:110 +0x1d8
perkeep.org/pkg/search_test.testQuery(...)
        /Users/bradfitz/src/perkeep.org/pkg/search/query_test.go:102
perkeep.org/pkg/search_test.TestQuery(0x14000bee000?)
        /Users/bradfitz/src/perkeep.org/pkg/search/query_test.go:183 +0x4c
testing.tRunner(0x14000bee000, 0x1008dcbf0)
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/testing/testing.go:1792 +0xe4
created by testing.(*T).Run in goroutine 1
        /Users/bradfitz/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.3.darwin-arm64/src/testing/testing.go:1851 +0x374
FAIL    perkeep.org/pkg/search  12.888s
FAIL
```

--- a/pkg/search/query_test.go
+++ b/pkg/search/query_test.go
@@ -156,6 +156,9 @@ func (qt *queryTest) wantRes(req *SearchQuery, wanted ...blob.Ref) {
 				qt.t.Fatalf("unexpected candidateSource: got %v, want %v", pickedCandidate, qt.candidateSource)
 			}
 		})
+		qt.t.Cleanup(func() {
+			ExportSetCandidateSourceHook(nil)
+		})
 	}
 	res, err := qt.Handler().Query(ctxbg, req)
 	if err != nil {

